src/tests.rs

```diff
@@ -158,6 +160,119 @@ fn test_sharing() {
 }
 
 #[test]
+fn test_multithread_render() {
+    load_gl();
+
+    let size = Size2D::new(256, 256);
+    let primary = GLContext::<NativeGLContext>::new(size,
+                                                    GLContextAttributes::default(),
+                                                    ColorAttachmentType::Texture,
+                                                    None).unwrap(); 
+    test_gl_context(&primary);
+    let (sender, receiver) = mpsc::channel();
+    thread::spawn(move ||{
+        //create the context in a different thread
+        let secondary = GLContext::<NativeGLContext>::new(size,
+                                                      GLContextAttributes::default(),
+                                                      ColorAttachmentType::Texture,
+                                                      None).unwrap();
+        secondary.make_current().unwrap();
+        assert!(secondary.is_current());
+        //render green adn test pixels
+        gl::clear_color(0.0, 1.0, 0.0, 1.0);
+        gl::clear(gl::COLOR_BUFFER_BIT);
+
+        let vec = gl::read_pixels(0, 0, size.width, size.height, gl::RGBA, gl::UNSIGNED_BYTE);
+        test_pixels_eq(&vec, &[0, 255, 0, 255]);
+
+        sender.send(()).unwrap();
+
+        // Avoid drop until test ends
+        loop {
+            thread::sleep(time::Duration::from_millis(200));
```

Comment:
This is nasty ;-)

Please add another kind of mechanism to block, another pair of channels would work.